tests/console/year_2038_detection.pm

```diff
@@ -34,43 +37,42 @@ sub install_pkg {
 }
 
 sub run {
+    my $self = shift;
     select_serial_terminal;
 
     install_pkg if (script_run('rpm -qi chrony') != 0);
     systemctl("start chronyd");    # Ensure chrony is started
-    assert_script_run 'chronyc makestep';
+    assert_script_run('chronyc makestep');
     record_info('Show current date and time', script_output('date +"%Y-%m-%d"'));
     assert_script_run('utmpdump /var/run/utmp');
     assert_script_run('utmpdump /var/log/wtmp');
 
     # Stop the chrony service so that we can change date and time
-    systemctl('stop chronyd.service');
+    disable_and_stop_service('chronyd.service');
+
     # Set the time and date beyond a Y2038
+    record_info('Set new date');
     assert_script_run('timedatectl set-time "2038-01-20 03:14:07"');
 
     # We may need to logout and login again to make the date/time change
-    # take effect. in a simple way, we can switch to another user to
-    # achieve this.
-    #
-    # Create user account, if image doesn't already contain user
-    # (which is the case for SLE images that were already prepared by openQA)
-    if (script_run("getent passwd $username") != 0) {
-        assert_script_run "useradd -m $testapi::username";
-        assert_script_run "echo '$testapi::username:$testapi::password' | chpasswd";
+    # However, we need to handle many platforms and different products.
+    # Reboot the system to acieve this is a simple way for the time being.
+    if (is_transactional) {
+        process_reboot(trigger => 1);
     }
-    ensure_serialdev_permissions;
-
-    # Switch user to check if the issue can be captured
-    select_console 'user-console';
+    else {
+        power_action('reboot', textmode => 1);
+        reconnect_mgmt_console if is_pvm;
+        $self->wait_boot(textmode => 1, ready_time => 600, bootloader_time => 300);
+    }
+    select_serial_terminal;
     my $utmp_output = script_output('utmpdump /var/run/utmp');
     my $wtmp_output = script_output('utmpdump /var/log/wtmp');
     record_soft_failure('bsc#1188626 uttmpdump shows incorrect year for 2038 and beyond') if ($utmp_output !~ m/2038/sx || $wtmp_output !~ m/2038/sx);
 
-    # Start the chrony service again
-    select_serial_terminal;
     systemctl('start chronyd.service');
-    assert_script_run 'chronyc makestep';
-    record_info('Show synced date and time', script_output('date +"%Y-%m-%d"'));
+    record_info('Show NTP sources', script_output('chronyc -n sources -v'));
+    script_retry('chronyc makestep && (date +"%Y-%m-%d" | grep -v 2038)', delay => 10, retry => 6, fail_message => 'Time sync with NTP server failed (poo#127343)');
```

Comment:
I would increase the delay in case of failures to at least 1 minute to let chrony cool down. This might help already. Then reduce also the number of attempts

```suggestion
    script_retry('chronyc makestep && (date +"%Y-%m-%d" | grep -v 2038)', delay => 60, retry => 3, fail_message => 'Time sync with NTP server failed (poo#127343)');
```